gr-blocks/lib/selector_impl.cc

```diff
@@ -0,0 +1,151 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2019 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * GNU Radio is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3, or (at your option)
+ * any later version.
+ *
+ * GNU Radio is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GNU Radio; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "selector_impl.h"
+#include <gnuradio/io_signature.h>
+#include <string.h>
+#include <stdexcept>
+
+namespace gr {
+namespace blocks {
+
+selector::sptr
+selector::make(size_t itemsize, unsigned int input_index, unsigned int output_index)
+{
+    return gnuradio::get_initial_sptr(
+        new selector_impl(itemsize, input_index, output_index));
+}
+
+selector_impl::selector_impl(size_t itemsize,
+                             unsigned int input_index,
+                             unsigned int output_index)
+    : block("selector",
+            io_signature::make(1, -1, itemsize),
+            io_signature::make(1, -1, itemsize)),
+      d_itemsize(itemsize),
+      d_enabled(true),
+      d_input_index(input_index),
+      d_output_index(output_index),
+      d_num_inputs(0),
+      d_num_outputs(0)
+{
+    message_port_register_in(pmt::mp("en"));
+    set_msg_handler(pmt::mp("en"), [this](pmt::pmt_t msg) { this->handle_enable(msg); });
```

Comment:
This will fail to compile on 3.7, even if it's so much more beautiful.
```suggestion
    set_msg_handler(pmt::mp("en"), boost::bind(&selector_impl::handle_enable, this _1));
```